Sources/Datadog/Core/System/NetworkPathMonitor.swift

```diff
@@ -0,0 +1,96 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Network
+
+@available(iOS 12, *)
+extension NWPathMonitor {
+    var current: NetworkConnectionInfo {
+        let info = currentPath
+
+        let availableInterfaces: [NetworkConnectionInfo.Interface] = {
+            info.availableInterfaces.map { interface in
+                switch interface.type {
+                case .wifi: return .wifi
+                case .wiredEthernet: return .wiredEthernet
+                case .cellular: return .cellular
+                case .loopback: return .loopback
+                case .other: return .other
+                @unknown default: return .other
+                }
+            }
+        }()
+
+        let reachability: NetworkConnectionInfo.Reachability = {
+            switch info.status {
+            case .satisfied: return .yes
+            case .requiresConnection: return .maybe
+            case .unsatisfied: return .no
+            @unknown default: return .maybe
+            }
+        }()
+
+        let isCurrentPathConstrained: Bool? = {
+            if #available(iOS 13.0, macOS 10.15, *) {
+                return info.isConstrained
+            } else {
+                return nil
+            }
+        }()
+
+        return NetworkConnectionInfo(
+            reachability: reachability,
+            availableInterfaces: availableInterfaces,
+            supportsIPv4: info.supportsIPv4,
+            supportsIPv6: info.supportsIPv6,
+            isExpensive: info.isExpensive,
+            isConstrained: isCurrentPathConstrained
+        )
+    }
+}
+
+// MARK: - Legacy Network Path Monitor
+
+import SystemConfiguration
+
+internal class LegacyPathMonitor {
```

Comment:
How about `iOS11PathMonitor`? That would serve cleaner purpose of this class.

Reply:
good idea 💯
it will be easier to find&remove when we drop iOS 11 support as well